packages/tools/client-debugger/client-debugger/src/messaging/AudienceMessages.ts

```diff
@@ -0,0 +1,30 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+// import { IAudience } from "@fluidframework/container-definitions";
+import { IClient } from "@fluidframework/protocol-definitions";
+import { AudienceChangeLogEntry } from "../Logs";
+import { IDebuggerMessage } from "./Messages";
+
+/**
+ * Message data format used by {@link AudienceEventMessage}.
+ * @public
+ */
+export interface AudienceEventMessageData {
+	/**
+	 * Contents of the Audience event
+	 */
+	audienceState: Map<string, IClient>;
```

Comment:
Should this also be `readonly`?

Reply:
We also probably need to convert our map to an array before transmitting this data, since `map`s aren't natively JSON-able. The keys in this map are the client IDs (for convenience), but those IDs can also be found in the IClient objects.

Reply:
We can update the visualizer to sort the array by client ID if we want to maintain stable ordering.

Reply:
Changed to `audienceClientMetaData: AudienceClientMetaData[]` which contains object with `clientId:string` and  `client:IClient`.